### PR TITLE
[MNG-7653] Integrate Resolver 2.x onSessionClose

### PR DESCRIPTION
**Pull request by cstamas**
Implement onSessionClose and "resolver 2.x way for sessions"...

As digging happens in same area, some other "resolver configuration" issues resolved as well.

---

https://issues.apache.org/jira/browse/MNG-7653
https://issues.apache.org/jira/browse/MNG-7935
https://issues.apache.org/jira/browse/MNG-7936

### CODE REVIEW

File: maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java [hunk 1]
```diff
@@ -100,4 +101,51 @@ public static DefaultRepositorySystemSession newSession() {
...
+        stereotypes.add(new DefaultArtifactType("ejb", "jar", "", "java"));
+        stereotypes.add(new DefaultArtifactType("ejb-client", "jar", "client", "java"));
+        stereotypes.add(new DefaultArtifactType("test-jar", "jar", "tests", "java"));
+        stereotypes.add(new DefaultArtifactType("javadoc", "jar", "javadoc", "java"));
+        stereotypes.add(new DefaultArtifactType("java-source", "jar", "sources", "java", false, false));
+        stereotypes.add(new DefaultArtifactType("war", "war", "", "java", false, true));
+        stereotypes.add(new DefaultArtifactType("ear", "ear", "", "java", false, true));
+        stereotypes.add(new DefaultArtifactType("rar", "rar", "", "java", false, true));
+        stereotypes.add(new DefaultArtifactType("par", "par", "", "java", false, true));
+        session.setArtifactTypeRegistry(stereotypes);
```
Comment (slawekjaranowski) [inline maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java:145]: Can you remind which issue / change requires it?

**Review by slawekjaranowski — APPROVED** [review]

File: maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java (hunk 1 above)
Comment (cstamas) [reply maven-resolver-provider/src/main/java/org/apache/maven/repository/internal/MavenRepositorySystemUtils.java:145]: Unsure what you mean specifically, but there is a related change to this area here: https://github.com/apache/maven/pull/1299

Today (and this is true for both old 3.x and master) there are two type registries coexisting in Maven: the resolver one (ArtifactTypeRegistry and legacy ArtifactHandlerManager). And we have introduced 3rd one in Maven4 API. The referenced PR "kills off" the coexistence, and makes only one (the new API) "in charge".

File: maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java [hunk 2]
```diff
@@ -308,6 +312,17 @@ public CLIManager() {
                 .optionalArg(true)
                 .desc("Defines the color mode of the output. Supported are 'auto', 'always', 'never'.")
                 .build());
+        options.addOption(Option.builder()
```
Comment (CrazyHZM) [inline maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java:315]: I suggest adding the abbreviated option `-canf`.

File: maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java [hunk 3]
```diff
@@ -308,6 +312,17 @@ public CLIManager() {
...
+                .longOpt(CACHE_ARTIFACT_NOT_FOUND)
+                .hasArg()
+                .desc(
+                        "Defines caching behaviour for 'not found' artifacts. Supported values are 'true' (default), 'false'.")
+                .build());
+        options.addOption(Option.builder()
```
Comment (CrazyHZM) [inline maven-embedder/src/main/java/org/apache/maven/cli/CLIManager.java:321]: I suggest adding the abbreviated option `-sadp`.

**Review by slawekjaranowski — APPROVED** [review]

**Comment by slawekjaranowski** [comment]
@cstamas please look at ITs it takes a very long time and look like can not be finished ....

**Review by gnodet — APPROVED** [review]

**Comment by jira-importer** [comment]
Resolve #8799

**Pull request merged**